platform/android/MapLibreAndroid/src/vulkan/java/org/maplibre/android/maps/renderer/textureview/TextureViewRenderThread.java

```diff
@@ -0,0 +1,271 @@
+package org.maplibre.android.maps.renderer.textureview;
+
+import android.graphics.SurfaceTexture;
+import android.view.Surface;
+import android.view.TextureView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.UiThread;
+
+import java.util.ArrayList;
+
+/**
+ * The render thread is responsible for managing the communication between the
+ * ui thread and the render thread it creates
+ */
+class TextureViewRenderThread extends Thread implements TextureView.SurfaceTextureListener {
```

Comment:
Most code in this class is identical to the GL version. They can probably use a common base class.

Reply:
Lifecycle function moved to shared code